### PR TITLE
Writing a NuGet.config as part of the preparatory steps

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

File: build/NugetConfigFile.targets [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+      <NugetConfigHeader>
+        &lt;?xml version=&quot;1.0&quot; encoding=&quot;utf-8&quot;?&gt;
+        &lt;configuration&gt;
+        &lt;packageSources&gt;
+        &lt;!--To inherit the global NuGet package sources remove the &lt;clear/&gt; line below --&gt;
+        &lt;clear /&gt;
+      </NugetConfigHeader>
+
+      <NugetConfigFeedsPrivate>
+        &lt;add key=&quot;PrivateBlobFeed&quot; value=&quot;$(ExternalRestoreSources)&quot; /&gt;
```
Comment (dsplaisted) [inline build/NugetConfigFile.targets]: Can there be more than one external restore source?  If so, then this probably won't work.  If not, then the variable seems named incorrectly.

File: run-build.ps1 [hunk 2]
```diff
@@ -106,7 +106,7 @@ if ($NoBuild)
 }
 else
 {
-    dotnet msbuild build.proj /p:Architecture=$Architecture /p:GeneratePropsFile=true /t:WriteDynamicPropsToStaticPropsFiles
+    dotnet msbuild build.proj /p:Architecture=$Architecture /p:GeneratePropsFile=true /t:WriteDynamicPropsToStaticPropsFiles $ExtraParameters
```
Comment (dsplaisted) [inline run-build.ps1]: I think it was probably intentional that the extra parameters weren't passed to this invocation.  I'd suggest asking Livar about this.

File: build/NugetConfigFile.targets [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+      </NugetConfigFeedsPrivate>
+      
+      <NugetConfigFeedsNormal>
+        &lt;add key=&quot;BlobFeed&quot; value=&quot;https://dotnetfeed.blob.core.windows.net/dotnet-core/index.json&quot; /&gt;
```
Comment (dsplaisted) [inline build/NugetConfigFile.targets]: You can probably make this prettier using `CDATA`, so you don't have to escape all the brackets.  See an example here: https://stackoverflow.com/a/2784200

**Review by mmitche — CHANGES_REQUESTED** [review]
Seems reasonable, assuming the multiple restore sources issue is fixed.

**Comment by johnbeisner** [comment]
@mmitche
@dsplaisted 
This should be finished now, please confirm.

**Review by mmitche — APPROVED** [review]

File: run-build.ps1 (hunk 2 above)
Comment (livarcocc) [reply run-build.ps1]: I don't remember why. But why do we need the extra parameters here? For the Extra restore sources? Should we make it explicit then?

File: run-build.ps1 [hunk 4]
```diff
@@ -80,16 +80,16 @@ $env:VSTEST_TRACE_BUILD=1
 # install a stage0
 $dotnetInstallPath = Join-Path $RepoRoot "scripts\obtain\dotnet-install.ps1"
 
-Write-Output "$dotnetInstallPath -Channel ""release/2.0.0"" -InstallDir $env:DOTNET_INSTALL_DIR -Architecture ""$Architecture"""
-Invoke-Expression "$dotnetInstallPath -Channel ""release/2.0.0"" -InstallDir $env:DOTNET_INSTALL_DIR -Architecture ""$Architecture"""
+Write-Output "$dotnetInstallPath -InstallDir $env:DOTNET_INSTALL_DIR -Architecture ""$Architecture"" -Channel ""release/2.0.0"""
```
Comment (livarcocc) [inline run-build.ps1]: Why the change here?

File: run-build.ps1 (hunk 2 above)
Comment (johnbeisner) [reply run-build.ps1]: The extra parameters needed in the 'GeneratePropsFile' step are: /p:ExternalRestoreSources & /p:DotNetBuildOffline - they are needed to control the content of the NuGet.config
When reviewing the properties now available to the 'GeneratePropsFile' step; I do not believe that the behavior of this pre-step would change.

File: run-build.ps1 (hunk 4 above)
Comment (johnbeisner) [reply run-build.ps1]: It's easier to read.

**Review by livarcocc — APPROVED** [review]

**Review by livarcocc — APPROVED** [review]
